### PR TITLE
added filter for access zone

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit tests
- [ ] Acceptance tests

### CODE REVIEW

File: docs/data-sources/accesszone.md [hunk 1]
```diff
@@ -88,7 +88,21 @@ output "powerscale_accesszone" {
...
 
-- `names` (Set of String)
+- `alternate_system_provider` (Set of String) List of alternate_system_provider
+- `cache_entry_expiry` (Set of Number) List of cache_entry_expiry
+- `create_path` (Boolean) Value for create_path
+- `groupnet` (Set of String) List of groupnet
+- `home_directory_umask` (Set of Number) List of home_directory_umask
+- `id` (Set of String) List of id
+- `map_untrusted` (Set of String) List of map_untrusted
+- `name` (Set of String) List of name
```
Comment (rounak-adhikary) [inline docs/data-sources/accesszone.md:98]: We are renaming a field here, which will break backward compatibility.
Instead, I suggest adding logic to convert plural names to singular names (remove the last `s` in list types) while filtering.

Comment (shenda1) [reply docs/data-sources/accesszone.md:98]: We can deprecate attribute https://developer.hashicorp.com/terraform/plugin/framework/deprecations#provider-attribute-rename

File: powerscale/models/access_zone.go [hunk 2]
```diff
@@ -80,7 +80,21 @@ type V1AuthAccessAccessItemFileGroup struct {
...
+	HomeDirectoryUmask       []types.Int64  `tfsdk:"home_directory_umask"`
+	ID                       []types.String `tfsdk:"id"`
+	MapUntrusted             []types.String `tfsdk:"map_untrusted"`
+	NegativeCacheEntryExpiry []types.Int64  `tfsdk:"negative_cache_entry_expiry"`
+	NetbiosName              []types.String `tfsdk:"netbios_name"`
+	Path                     []types.String `tfsdk:"path"`
+	SkeletonDirectory        []types.String `tfsdk:"skeleton_directory"`
+	SystemProvider           []types.String `tfsdk:"system_provider"`
+	ZoneID                   []types.Int64  `tfsdk:"zone_id"`
+	CreatePath               types.Bool     `tfsdk:"create_path"`
```
Comment (baoy1) [inline powerscale/models/access_zone.go:96]: Please check if create_path is included in real response. I think it is used by POST, though it is listed in the REST GET schema.

**Review by baoy1 — APPROVED** [review]

**Comment by Krishnan-Priyanshu** [comment]
As we will be support fliter for API Query Parameter, will be closing this one.

**Pull request closed without merging**